scipy/signal/_ltisys.py

```diff
@@ -2098,10 +2098,6 @@ def lsim(system, U, T, X0=None, interp=True):
         return T, squeeze(yout), squeeze(xout)
 
     dt = T[1] - T[0]
-    if not np.allclose((T[1:] - T[:-1]) / dt, 1.0):
-        warnings.warn("Non-uniform timesteps are deprecated. Results may be "
-                      "slow and/or inaccurate.", DeprecationWarning)
-        return lsim2(system, U, T, X0)
```

Comment:
This should raise an error now to enforce the docstring 
```
    T : array_like
        The time steps at which the input is defined and at which the
        output is desired.  Must be nonnegative, increasing, and equally spaced.
```

The errors should also point out something like "You can use `lsim2(system, U, T, X0)` for non-uniform timesteps, though the results may be slow and/or inaccurate".

Reply:
Actually I just realized that this conflicts with the goals of #5676, so this will need a bit of discussion first - sorry for that.

Reply:
Yeah, unfortunately it seems that we'll have to tackle this differently (and it's going to be more invasive/finnicky, so no expectation that you do it in this PR or at all). See this [comment](https://github.com/scipy/scipy/issues/15755#issuecomment-1072799430) in the issue.